### DISCUSSION TITLE
How to debug/trace auto-renewal process of certificates

### DISCUSSION THREAD

**Discussion by schoenr79** [polymind-inc/acmebot#363] [Q&A] [answered]
Hi,

I'm looking for a way to trace/debug the auto-renewal process of existing certificates. because last time a renewal wasn"t triggered, so the cert expired.

- Is there a way how to debug and trace this process
- is it possible to tirgger a cert-renew manual? / API-Call etc.

best regards
Roland

**Comment by shibayan** [comment, marked as answer]
Acmebot is the same as regular Azure Functions, so you can trace and debug it from Application Insights.

Manual renewal can be done on a per-certificate basis via `/renew-certificate`, or you can manually trigger the function `RenewCertificates_Timer` to renew all certificates that are close to expiring.

In rare cases, Azure Functions timer triggers may not be executed, in which case, restarting from Azure Portal will improve the situation.
